src/bin/commands/demux.rs

```diff
@@ -239,7 +245,11 @@ impl ReadSet {
                         &chars[first_colon_idx + 1..chars.len()]
                     };
 
-                    write!(writer, "{}:", read_num)?;
+                    if read_num < Self::READ_NUMBERS.len() {
```

Comment:
If you don't take the suggestion above about LazyLock and starting at 0.
```suggestion
                    if read_num <=Self::READ_NUMBERS.len() {
```